### DISCUSSION TITLE
기존 데이터베이스 스키마가 변경될 경우

### DISCUSSION THREAD

**Discussion by rlajm1203** [JNU-econovation/EEOS-BE#160] [Announcements]
EEOS 3.0 버전을 배포하는 과정에서, 데이터베이스의 스키마 구조가 변경되었습니다.

Program 테이블에 새로운 컬럼 하나가 추가 되고, Team 테이블, Presentation 테이블이 추가되었습니다.

이렇게 기존에 운영하던 데이터베이스의 스키마 구조가 변경이 되었을 때, 기존에 존재하던 테이블에서 새로 추가된 컬럼의 값을 NULL로 유지할지, 아니면 적당한 디폴트 값을 채워 넣을지 고민이 들었습니다.

저의 생각은, "데이터베이스에 NULL 값을 넣도록 유지하는 것은 옳지 않다."입니다. 그 이유는 NULL이라는 값 자체가 데이터가 없음을 의미할 수도 있고 개발자의 의도에 따라서 다르게 해석될 수 있기 때문입니다.

그래서 어떤 값이든 디폴트 값을 정해서 저장을 해야 한다고 생각하는데 다른 분들께서는 어떻게 생각하시나요?

**Comment by kssumin** [comment]
디폴트 값을 정해서 해당 값으로 채워야 한다고 생각합니다.
NULL이 들어있으면 이번처럼 NULL에 대한 별도의 처리를 하지 않았을 경우 에러가 터지는데 이걸 대체하는 것보다는
기획적으로 디폴트가 무엇인지를 의견나누는 게 좋을 것 같다는 생각입니다.

또한, 사실 저희가 개발을 하다보면 기존 테이블에 칼럼이 추가되었는지 여부를 배포할 때는 까먹는 케이스가 더 많다라고 생각합니다.
추가적으로 배포를 한 이후에 데이터 작업(NULL이였던 걸 디폴트로 채우는 작업)을 해야할텐데
이 두 작업 사이에는 사이트가 다운된다는 문제도 있는 것 같습니다.

그래서 저는 적극적으로 데이터베이스 스키마 관리 툴인 flyway를 사용하자고 의견드립니다!

Flyway사용하면 배포 시에 데이터관련 작업도 같이 처리
